libs/ui/src/screens/send/screens/send-token/components/token-input/token-input.tsx

```diff
@@ -0,0 +1,124 @@
+import { isDefined, isNotEmptyString } from '@rnw-community/shared';
+import React from 'react';
+import { Controller, RegisterOptions } from 'react-hook-form';
+import { FieldPath } from 'react-hook-form/dist/types';
+import { FieldValues } from 'react-hook-form/dist/types/fields';
+import { Control, UseFormSetValue, UseFormTrigger } from 'react-hook-form/dist/types/form';
+import { View } from 'react-native';
+
+import { Button } from '../../../../../../components/button/button';
+import { ButtonThemesEnum } from '../../../../../../components/button/enums';
+import { Text } from '../../../../../../components/text/text';
+import { Label } from '../../../../../../components/text-input/components/label/label';
+import { TextInput } from '../../../../../../components/text-input/text-input';
+import { Token as TokenType } from '../../../../../../interfaces/token.interface';
+
+import { DollarAmount } from './components/dollar-amount/dollar-amount';
+import { SelectToken } from './components/select-token/select-token';
+import { styles } from './token-input.styles';
+
+interface Props<TFieldValues extends FieldValues, TName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>> {
+  label: string;
+  control: Control<TFieldValues>;
+  setValue?: UseFormSetValue<TFieldValues>;
+  trigger?: UseFormTrigger<TFieldValues>;
+  rules?: Pick<RegisterOptions<TFieldValues>, 'required'>;
+  error?: string;
+  token?: TokenType;
+  amountInDollar: string;
+  name: TName;
+  isReadOnly?: boolean;
+  tokenParam: string;
+  amount: string;
+  availableFormattedBalance?: string;
+  availableBalance?: string;
+  selectTokensWithBalance?: boolean;
+}
+
+// eslint-disable-next-line  @typescript-eslint/no-explicit-any
+export const TokenInput = <TType extends FieldValues & Record<string, any>>({
```

Comment:
find the way how correctly to make shared form filed components @MuKoJla @svjatoslavtt @Boltman92